src/cpp/fastdds/core/policy/QosPoliciesSerializer.hpp

```diff
@@ -1240,6 +1245,645 @@ inline bool QosPoliciesSerializer<TopicDataQosPolicy>::read_from_cdr_message(
                    parameter_length);
 }
 
+template<>
+inline uint32_t QosPoliciesSerializer<RTPSEndpointQos>::cdr_serialized_size(
+        const RTPSEndpointQos& qos_policy)
+{
+    // p_id + p_length
+    uint32_t ret_val = 2 + 2;
+
+    // + unicast locator list size
+    ret_val += 4;
+    for (rtps::LocatorListConstIterator it = qos_policy.unicast_locator_list.begin();
+            it != qos_policy.unicast_locator_list.end();
+            ++it)
+    {
+        // kind + port + address
+        ret_val += 4 + 4 + 16;
+    }
+
+    // + multicast locator list size
+    ret_val += 4;
+    for (rtps::LocatorListConstIterator it = qos_policy.multicast_locator_list.begin();
+            it != qos_policy.unicast_locator_list.end();
+            ++it)
+    {
+        // kind + port + address
+        ret_val += 4 + 4 + 16;
+    }
+
+    // + remote locator list
+    ret_val += 4;
+    for (rtps::LocatorListConstIterator it = qos_policy.remote_locator_list.begin();
+            it != qos_policy.unicast_locator_list.end();
+            ++it)
+    {
+        // kind + port + address
+        ret_val += 4 + 4 + 16;
+    }
+
+    // Do not serialize external_locators yet.
```

Comment:
Might be a good idea to add a length = 0 here, to leave room for a future implementation:
```suggestion
    // Do not serialize external_locators yet, but leave a length = 0 field here.
    ret_val += 4;
```